### PR TITLE
Shell executor logs through testing.T in upgrade tests

### PR DESCRIPTION
**Pull request by mgencur**
Similar to https://github.com/knative/serving/pull/14495

## Proposed Changes

- Print logs from shell executor to testing.T so that the logs can be found within test scope

**Release Note**

```release-note

```

**Docs**

### CODE REVIEW

**Comment by mgencur** [comment]
Printing within test scope can be seen in [this run](https://prow.knative.dev/view/gs/knative-prow/pr-logs/pull/knative-extensions_eventing-kafka-broker/3398/upgrade-tests_eventing-kafka-broker_main/1713843287481126912)
Now removing the temporary commit to make the tests pass.

**Comment by mgencur** [comment]
Unrelated failure. Re-running.

/test reconciler-tests-namespaced-broker

File: control-plane/pkg/reconciler/consumergroup/consumergroup_test.go
Comment (creydr) [inline control-plane/pkg/reconciler/consumergroup/consumergroup_test.go:1]: This should have been fixed in #3422. Lets see if it requires a rebase...

**Review by creydr — APPROVED** [review]
seems LGTM
/lgtm

**Pull request merged**